plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -713,7 +714,10 @@ public static OptionalInt parseVersion(String metadataLocation)
         int versionStart = metadataLocation.lastIndexOf('/') + 1; // if '/' isn't found, this will be 0
         int versionEnd = metadataLocation.indexOf('-', versionStart);
         if (versionStart == 0 || versionEnd == -1) {
-            throw new TrinoException(ICEBERG_BAD_DATA, "Invalid metadata location: " + metadataLocation);
+            String metadataFileName = metadataLocation.substring(versionStart);
```

Comment:
update the code to use `io.trino.filesystem.Location#fileName` as a follow-up.

Reply:
done in https://github.com/trinodb/trino/pull/18150. Due to other changes in that PR, I think it would be better to merge https://github.com/trinodb/trino/pull/18150 before merging this one. Please consider rebasing.